File: 백준/Silver/1991. 트리 순회/트리 순회.py
def preorder(root):
    print(root, end="")
    if binary_tree[root][0] != ".":
        preorder(binary_tree[root][0])
    if binary_tree[root][1] != ".":
        preorder(binary_tree[root][1])

def inorder(root):

    if binary_tree[root][0] != ".":
        inorder(binary_tree[root][0])
    print(root, end="")
    if binary_tree[root][1] != ".":
        inorder(binary_tree[root][1])

def postorder(root):

    if binary_tree[root][0] != ".":
        postorder(binary_tree[root][0])
    if binary_tree[root][1] != ".":
        postorder(binary_tree[root][1])
    print(root, end="")

N = int(input())

binary_tree = {}
root = None

for i in range(N):
    tmp = input().split()
    if(i == 0):
        root = tmp[0]
    binary_tree[tmp[0]]=[tmp[1],tmp[2]]

preorder(root)
print()
inorder(root)
print()
postorder(root)

File: 백준/Silver/1991. 트리 순회/트리 순회.java
import com.sun.source.tree.Tree;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;

class Main {
    static class TreeNode {
        char data;
        char left;
        char right;

        public TreeNode(char data, char left, char right) {
            this.data = data;
            this.left = left;
            this.right = right;
        }
    }
    public static void preOrder(TreeNode root){

        System.out.print(root.data);

        if(root.left != '.'){
            preOrder(tree.get(root.left));
        }

        if(root.right != '.'){
            preOrder(tree.get(root.right));
        }
    }

    public static void inOrder(TreeNode root){

        if(root.left != '.'){
            inOrder(tree.get(root.left));
        }

        System.out.print(root.data);

        if(root.right != '.'){
            inOrder(tree.get(root.right));
        }
    }

    public static void postOrder(TreeNode root){

        if(root.left != '.'){
            postOrder(tree.get(root.left));
        }

        if(root.right != '.'){
            postOrder(tree.get(root.right));
        }

        System.out.print(root.data);
    }

    static Map<Character,TreeNode> tree = new HashMap<>();

    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            char a = st.nextToken().charAt(0);
            char b = st.nextToken().charAt(0);
            char c = st.nextToken().charAt(0);

            tree.put(a, new TreeNode(a, b, c));
        }

        preOrder(tree.get('A'));
        System.out.println();
        inOrder(tree.get('A'));
        System.out.println();
        postOrder(tree.get('A'));
    }
}
